### PR TITLE
JAMES-2949 Varous server/data refactorings

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/data/data-jpa/src/main/java/org/apache/james/domainlist/jpa/JPADomainList.java [hunk 1]
```diff
@@ -79,41 +75,30 @@ public void init() {
...
-        return ImmutableList.copyOf(domains);
     }
 
     @Override
     protected boolean containsDomainInternal(Domain domain) throws DomainListException {
         EntityManager entityManager = entityManagerFactory.createEntityManager();
-        final EntityTransaction transaction = entityManager.getTransaction();
         try {
-            transaction.begin();
-            boolean result = containsDomainInternal(domain, entityManager);
```
Comment (mbaechler) [inline server/data/data-jpa/src/main/java/org/apache/james/domainlist/jpa/JPADomainList.java]: It's probably a problem: a transaction also mean you will have a consistent view for reads during the transaction. It's the isolation feature of the MVCC model.

Comment (chibenwa) [reply server/data/data-jpa/src/main/java/org/apache/james/domainlist/jpa/JPADomainList.java]: I don t think so.

You have the guaranty to not see partially applied transacrions. As long as we get a single read statement, we don't need to explicitly wrap it in a transaction.

See https://stackoverflow.com/questions/26327274/do-you-need-a-database-transaction-for-reading-data

**Comment by Arsnael** [comment]
```
[d784b152487af8f55f653c8d7e56a063fb272289] [ERROR] Errors: 
[d784b152487af8f55f653c8d7e56a063fb272289] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[d784b152487af8f55f653c8d7e56a063fb272289] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[d784b152487af8f55f653c8d7e56a063fb272289] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
[d784b152487af8f55f653c8d7e56a063fb272289] [ERROR]   CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution
```
test this please

File: server/data/data-jpa/src/main/java/org/apache/james/user/jpa/JPAUsersRepository.java [hunk 2]
```diff
@@ -98,23 +98,6 @@ public User getUserByName(Username name) throws UsersRepositoryException {
...
-    /**
-     * Returns the user name of the user matching name on an equalsIgnoreCase
-     * basis. Returns null if no match.
-     * 
-     * @param name
-     *            the name to case-correct
-     * @return the case-correct name of the user, null if the user doesn't exist
-     * @throws UsersRepositoryException
-     */
-    public String getRealName(Username name) throws UsersRepositoryException {
```
Comment (rouazana) [inline server/data/data-jpa/src/main/java/org/apache/james/user/jpa/JPAUsersRepository.java]: could be used to get the case sensitive version of a username store in lower case?

Comment (chibenwa) [reply server/data/data-jpa/src/main/java/org/apache/james/user/jpa/JPAUsersRepository.java]: It's not part of the API

And a change for using similar mechanisms (but at the lib level) is under discussion here: https://github.com/linagora/james-project/pull/2887

I don't think this method adds value

**Comment by chibenwa** [comment]
Merged

File: server/data/data-jpa/src/main/java/org/apache/james/domainlist/jpa/JPADomainList.java (hunk 1 above)
Comment (mbaechler) [reply server/data/data-jpa/src/main/java/org/apache/james/domainlist/jpa/JPADomainList.java]: > I don t think so.

You answer to which statement?

> You have the guaranty to not see partially applied transacrions. As long as we get a single read statement, we don't need to explicitly wrap it in a transaction.

I agree with you concerning single statements: It doesn't make a difference. However, I still claim transaction give you isolation (if you configure it that way, see https://stackoverflow.com/questions/16162357/transaction-isolation-levels-relation-with-locks-on-table). BTW, we should use at least "REPEATABLE READ".

**Pull request closed without merging**